experimental/dds/tree2/src/feature-libraries/editable-tree/editableField.ts

```diff
@@ -47,13 +48,34 @@ import {
 import { makeTree } from "./editableTree";
 import { ProxyTarget } from "./ProxyTarget";
 
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-bitwise */
+
+/**
+ * Coerces the given 'index' to an Int32.  This includes the following behaviors:
+ *
+ * - +/-Infinity is preserved
```

Comment:
Why preserve infinity? This is an index into a collection of known size: I don't think infinity makes sense in this context and I think having it error would be better.

Generally we take the approach of having malformed input error (fail fast) rather than try and recover with some plausible interpretation, and this whole function goes against that in many ways.